src/logic/community/types.ts

```diff
@@ -3,7 +3,11 @@ import { CommunityRole, Pagination } from '../../types/entities'
 
 export type ICommunityComponent = {
   getCommunity: (id: string, userAddress: string) => Promise<CommunityWithMembersCount>
-  getCommunities: (userAddress: string, options: GetCommunitiesOptions) => Promise<GetCommunitiesResult>
+  getCommunities: (
+    userAddress: string,
+    options: GetCommunitiesOptions
+  ) => Promise<GetCommunitiesResult<CommunityResult>>
```

Comment:
Super NIT: Result sounds like an HTTP response but here we are declaring the return from the logic component which should be HTTP agnostic 🤔 

```suggestion
  ) => Promise<GetCommunitiesResult<CommunityResult>>
```